### DISCUSSION TITLE
Error when enabling Yubikey

### DISCUSSION THREAD

**Discussion by freud44** [gravitational/teleport#31089] [Q&A]
Hello
I'm using Teleport for testing with docker, so no certs use for the moment.
I'm trying to enabling a yubikey on an account, after verifying the user, when I'm enabling the hardware key, I'm having an error message "this is not the righ domain".

As I'm not using a certificate, is this related?

**Comment by zmb3** [comment]
Did you set `rp_id` in your auth server's `teleport.yaml`?

**Comment by freud44** [comment]
Yes, I put localhost as value, and trying also to use the server's proxy IP address.
But none of those 2 values worked.

**Comment by freud44** [comment]
is it mandatory to have a domain registered or certificate installed in order to use a yubikey ?

**Reply by webvictim** [reply]
It's not mandatory to have a domain registered (you can use `1.2.3.4.nip.io` for example, replacing `1.2.3.4` with your own IP address) but you **do** need to get a trusted TLS certificate valid for that domain, as webauthn will not work when there are TLS certificate issues in the browser. If you can get a valid TLS cert and set the domain under `rp_id`, webauthn/Yubikeys should work.

Some ways of doing that:
- run `certbot` to complete a challenge and get a TLS cert, then provide that to Teleport using `proxy_service.https_keypairs`
- set `proxy_service.acme.enabled` and `proxy_service.acme.email` in the Teleport config file, make sure port 443 accepts traffic from the internet (forward the port or similar)
- generate your own self-signed cert with `openssl` and trust the certificate at the browser/OS level
- put a load balancer or reverse proxy in front of Teleport with a trusted TLS certificate for Teleport's domain
